src/unaligned_vector/binary_quantized.rs

```diff
@@ -0,0 +1,299 @@
+    let output_ptr = output.as_mut_ptr();
+    let bytes = vec.as_bytes();
+    let low_mask = [0b_0000_0001, 0b_0000_0010, 0b_0000_0100, 0b_0000_1000];
+    let high_mask = [0b_0001_0000, 0b_0010_0000, 0b_0100_0000, 0b_1000_0000];
+    let ones = unsafe { _mm_set1_ps(1.0) };
+    let minus = unsafe { _mm_set1_ps(-1.0) };
+
+    for (current_byte, base) in bytes.iter().enumerate() {
+        unsafe {
+            let base = _mm_set1_epi32(*base as i32);
+            for (i, mask) in [low_mask, high_mask].iter().enumerate() {
+                let mask = _mm_set_epi32(mask[3], mask[2], mask[1], mask[0]);
+                let mask = _mm_and_si128(base, mask);
+                // 0xffffffff if equal to zero and 0x00000000 otherwise
+                let mask = _mm_cmpeq_epi32(mask, _mm_setzero_si128());
+                let lane = _mm_blendv_ps(ones, minus, _mm_castsi128_ps(mask));
+                let offset = output_ptr.add(current_byte * 8 + i * 4);
+                _mm_store_ps(offset, lane);
+            }
+        }
+    }
+
+    output
+}
+
+// Dedicated to mm256 (AVX). Doesn't provide any real perf gain.
+// #[cfg(not(any(target_arch = "aarch64", target_arch = "arm64ec")))]
+// unsafe fn to_vec_avx(vec: &UnalignedVector<BinaryQuantized>) -> Vec<f32> {
+//     use core::arch::x86_64::*;
+
+//     let mut output: Vec<f32> = vec![0.0; vec.len()];
+//     let output_ptr = output.as_mut_ptr();
+//     let bytes = vec.as_bytes();
+//     let mask = [
+//         0b_0000_0001,
+//         0b_0000_0010,
+//         0b_0000_0100,
+//         0b_0000_1000,
+//         0b_0001_0000,
+//         0b_0010_0000,
+//         0b_0100_0000,
+//         0b_1000_0000,
+//     ];
+//     let ones = unsafe { _mm256_set1_ps(1.0) };
+//     let minus = unsafe { _mm256_set1_ps(-1.0) };
+
+//     for (current_byte, base) in bytes.iter().enumerate() {
+//         unsafe {
+//             let base = _mm256_set1_epi32(*base as i32);
+//             let mask = _mm256_set_epi32(
+//                 mask[7], mask[6], mask[5], mask[4], mask[3], mask[2], mask[1], mask[0],
+//             );
+//             let mask = _mm256_and_si256(base, mask);
+//             // 0xffffffff if equal to zero and 0x00000000 otherwise
+//             let mask = _mm256_cmpeq_epi32(mask, _mm256_setzero_si256());
+//             let lane = _mm256_blendv_ps(ones, minus, _mm256_castsi256_ps(mask));
+//             let offset = output_ptr.add(current_byte * 8);
+//             // I don't understand why this is not aligned and I cannot use _mm256_store_ps
+//             _mm256_storeu_ps(offset, lane);
+//         }
+//     }
+
+//     output
+// }
+
+pub struct BinaryQuantizedIterator<'a> {
+    current_element: QuantizedWord,
+    current_iteration: usize,
+    iter: ChunksExact<'a, u8>,
+}
+
+impl Iterator for BinaryQuantizedIterator<'_> {
+    type Item = f32;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current_iteration >= QUANTIZED_WORD_SIZE {
```

Comment:
Rename it something like `QUANTIZED_WORD_BITS`

Reply:
Done in https://github.com/meilisearch/arroy/pull/82/commits/71f6659bb4aeed3ea64d536ba566034f66c0ad4d